In [2]:
from google.colab import drive
drive.mount('/content/drive')

import os

base_path = "/content/drive/MyDrive/contest_changwon_port"

trucking_folder = os.path.join(base_path, "data/raw/trucking_company")
os.makedirs(trucking_folder, exist_ok=True)

print("폴더 위치:", trucking_folder)

Mounted at /content/drive
폴더 위치: /content/drive/MyDrive/contest_changwon_port/data/raw/trucking_company


In [3]:
import pandas as pd
import glob
import os

files = glob.glob(os.path.join(trucking_folder, "*.csv"))

print("찾은 파일:")
for f in files:
    print(os.path.basename(f))

trucking_path = files[0]

trucking = pd.read_csv(trucking_path, encoding="cp949")

print("사용한 파일:", os.path.basename(trucking_path))
print("데이터 크기:", trucking.shape)
display(trucking.head())
print(trucking.columns)

찾은 파일:
경상남도 창원시_일반화물자동차운송사업자_20230620.csv
사용한 파일: 경상남도 창원시_일반화물자동차운송사업자_20230620.csv
데이터 크기: (566, 6)


,업체명,면허종류,차량대수(직영),차량대수(위수탁),차량대수(확인불가),주소
0,개인사업자(업체명없음),일반화물,1.0,NaN,NaN,"경상남도 창원시 의창구 서곡길 133 (봉곡동,늘푸른마을 코오롱아파트)"
1,(주)해솔로지스,일반화물,NaN,2.0,NaN,"경상남도 창원시 의창구 사림로 47, 사림프라자 504호 (사림동)"
2,(주)창진로지스,일반화물,NaN,2.0,NaN,"경상남도 창원시 의창구 차상로 58, 한신빌딩 2층 202호 (팔용동)"
3,개인사업자(업체명없음),일반화물,2.0,NaN,NaN,"경상남도 창원시 의창구 서상로 27 (동정동,동정오성아파트)"
4,(주)우경특수,일반화물,NaN,2.0,NaN,"경상남도 창원시 의창구 차룡로48번길 44, 창원스마트업타워 S1807호 (팔용동)"


Index(['업체명', '면허종류', '차량대수(직영)', '차량대수(위수탁)', '차량대수(확인불가)', '주소'], dtype='object')


In [5]:
# 구별 운송사업자 및 차량대수

import pandas as pd
import os

base_path = "/content/drive/MyDrive/contest_changwon_port"

trucking_path = os.path.join(
    base_path,
    "data/raw/trucking_company",
    "경상남도 창원시_일반화물자동차운송사업자_20230620.csv"
)

trucking = pd.read_csv(trucking_path, encoding="cp949")

print("데이터 크기:", trucking.shape)
display(trucking.head())
print(trucking.columns)

데이터 크기: (566, 6)


,업체명,면허종류,차량대수(직영),차량대수(위수탁),차량대수(확인불가),주소
0,개인사업자(업체명없음),일반화물,1.0,NaN,NaN,"경상남도 창원시 의창구 서곡길 133 (봉곡동,늘푸른마을 코오롱아파트)"
1,(주)해솔로지스,일반화물,NaN,2.0,NaN,"경상남도 창원시 의창구 사림로 47, 사림프라자 504호 (사림동)"
2,(주)창진로지스,일반화물,NaN,2.0,NaN,"경상남도 창원시 의창구 차상로 58, 한신빌딩 2층 202호 (팔용동)"
3,개인사업자(업체명없음),일반화물,2.0,NaN,NaN,"경상남도 창원시 의창구 서상로 27 (동정동,동정오성아파트)"
4,(주)우경특수,일반화물,NaN,2.0,NaN,"경상남도 창원시 의창구 차룡로48번길 44, 창원스마트업타워 S1807호 (팔용동)"


Index(['업체명', '면허종류', '차량대수(직영)', '차량대수(위수탁)', '차량대수(확인불가)', '주소'], dtype='object')


In [7]:
# 차량대수 숫자 처리

vehicle_cols = [
    "차량대수(직영)",
    "차량대수(위수탁)",
    "차량대수(확인불가)"
]

for col in vehicle_cols:
    trucking[col] = pd.to_numeric(trucking[col], errors="coerce").fillna(0)

trucking["총차량대수"] = (
    trucking["차량대수(직영)"] +
    trucking["차량대수(위수탁)"] +
    trucking["차량대수(확인불가)"]
)

display(trucking.head())

,업체명,면허종류,차량대수(직영),차량대수(위수탁),차량대수(확인불가),주소,총차량대수
0,개인사업자(업체명없음),일반화물,1.0,0.0,0.0,"경상남도 창원시 의창구 서곡길 133 (봉곡동,늘푸른마을 코오롱아파트)",1.0
1,(주)해솔로지스,일반화물,0.0,2.0,0.0,"경상남도 창원시 의창구 사림로 47, 사림프라자 504호 (사림동)",2.0
2,(주)창진로지스,일반화물,0.0,2.0,0.0,"경상남도 창원시 의창구 차상로 58, 한신빌딩 2층 202호 (팔용동)",2.0
3,개인사업자(업체명없음),일반화물,2.0,0.0,0.0,"경상남도 창원시 의창구 서상로 27 (동정동,동정오성아파트)",2.0
4,(주)우경특수,일반화물,0.0,2.0,0.0,"경상남도 창원시 의창구 차룡로48번길 44, 창원스마트업타워 S1807호 (팔용동)",2.0


In [8]:
trucking["구"] = trucking["주소"].astype(str).str.extract(r"창원시\s*([^\s]+구)")[0]

print(trucking["구"].unique())
display(trucking[["업체명", "주소", "구"]].head())

['의창구' '성산구' '마산합포구' '마산회원구' '진해구']


,업체명,주소,구
0,개인사업자(업체명없음),"경상남도 창원시 의창구 서곡길 133 (봉곡동,늘푸른마을 코오롱아파트)",의창구
1,(주)해솔로지스,"경상남도 창원시 의창구 사림로 47, 사림프라자 504호 (사림동)",의창구
2,(주)창진로지스,"경상남도 창원시 의창구 차상로 58, 한신빌딩 2층 202호 (팔용동)",의창구
3,개인사업자(업체명없음),"경상남도 창원시 의창구 서상로 27 (동정동,동정오성아파트)",의창구
4,(주)우경특수,"경상남도 창원시 의창구 차룡로48번길 44, 창원스마트업타워 S1807호 (팔용동)",의창구


In [10]:
# 구별 업체 수, 차량대수 집계

trucking_gu_summary = (
    trucking
    .groupby("구", as_index=False)
    .agg(
        업체수=("업체명", "count"),
        직영차량대수=("차량대수(직영)", "sum"),
        위수탁차량대수=("차량대수(위수탁)", "sum"),
        확인불가차량대수=("차량대수(확인불가)", "sum"),
        총차량대수=("총차량대수", "sum")
    )
)

trucking_gu_summary["업체비중"] = (
    trucking_gu_summary["업체수"] / trucking_gu_summary["업체수"].sum() * 100
)

trucking_gu_summary["차량비중"] = (
    trucking_gu_summary["총차량대수"] / trucking_gu_summary["총차량대수"].sum() * 100
)

trucking_gu_summary["업체당_평균차량대수"] = (
    trucking_gu_summary["총차량대수"] / trucking_gu_summary["업체수"]
)

trucking_gu_summary = trucking_gu_summary.sort_values(
    "총차량대수",
    ascending=False
)

display(trucking_gu_summary)

,구,업체수,직영차량대수,위수탁차량대수,확인불가차량대수,총차량대수,업체비중,차량비중,업체당_평균차량대수
1,마산회원구,121,161.0,776.0,131.0,1068.0,21.378092,30.566686,8.826446
3,의창구,134,265.0,376.0,246.0,887.0,23.674912,25.386377,6.619403
4,진해구,135,266.0,158.0,156.0,580.0,23.851590,16.599886,4.296296
2,성산구,103,127.0,295.0,84.0,506.0,18.197880,14.481969,4.912621
0,마산합포구,73,153.0,221.0,79.0,453.0,12.897527,12.965083,6.205479


In [11]:
# 항만 인접지역 / 내륙 산단 배후 지역 비교

def classify_area(gu):
    if gu in ["마산합포구", "진해구"]:
        return "항만 인접지역"
    else:
        return "내륙·산단 배후지역"

trucking["권역구분"] = trucking["구"].apply(classify_area)

trucking_area_summary = (
    trucking
    .groupby("권역구분", as_index=False)
    .agg(
        업체수=("업체명", "count"),
        총차량대수=("총차량대수", "sum")
    )
)

trucking_area_summary["업체비중"] = (
    trucking_area_summary["업체수"] / trucking_area_summary["업체수"].sum() * 100
)

trucking_area_summary["차량비중"] = (
    trucking_area_summary["총차량대수"] / trucking_area_summary["총차량대수"].sum() * 100
)

trucking_area_summary["업체당_평균차량대수"] = (
    trucking_area_summary["총차량대수"] / trucking_area_summary["업체수"]
)

display(trucking_area_summary)

,권역구분,업체수,총차량대수,업체비중,차량비중,업체당_평균차량대수
0,내륙·산단 배후지역,358,2461.0,63.250883,70.435031,6.874302
1,항만 인접지역,208,1033.0,36.749117,29.564969,4.966346


In [12]:
license_summary = (
    trucking
    .groupby(["구", "면허종류"], as_index=False)
    .agg(
        업체수=("업체명", "count"),
        총차량대수=("총차량대수", "sum")
    )
    .sort_values(["구", "총차량대수"], ascending=[True, False])
)

display(license_summary)

,구,면허종류,업체수,총차량대수
1,마산합포구,일반화물,5,338.0
0,마산합포구,(구)일반화물,68,115.0
2,마산회원구,(구)일반화물,114,995.0
3,마산회원구,일반화물,7,73.0
4,성산구,(구)일반화물,58,262.0
5,성산구,일반화물,45,244.0
6,의창구,(구)일반화물,113,554.0
7,의창구,일반화물,21,333.0
8,진해구,(구)일반화물,121,537.0
9,진해구,일반화물,14,43.0


In [13]:
processed_path = os.path.join(base_path, "data/processed")
os.makedirs(processed_path, exist_ok=True)

trucking_gu_summary.to_csv(
    os.path.join(processed_path, "trucking_gu_summary_2023.csv"),
    index=False,
    encoding="utf-8-sig"
)

trucking_area_summary.to_csv(
    os.path.join(processed_path, "trucking_area_summary_2023.csv"),
    index=False,
    encoding="utf-8-sig"
)

license_summary.to_csv(
    os.path.join(processed_path, "trucking_license_summary_2023.csv"),
    index=False,
    encoding="utf-8-sig"
)

trucking.to_csv(
    os.path.join(processed_path, "trucking_processed_2023.csv"),
    index=False,
    encoding="utf-8-sig"
)

print("일반화물자동차운송사업자 분석 결과 저장 완료")

일반화물자동차운송사업자 분석 결과 저장 완료


일반화물자동차운송사업자 현황 분석 결과, 창원시에는 총 566개의 일반화물자동차 운송사업자가 확인, 구별로는 진해구 135개, 의창구 134개, 마산회원구 121개, 성산구 103개, 마산합포구 73개 순으로 나타남, 그러나 차량대수 기준으로는 마산회원구가 1,068대로 가장 많았고, 의창구 887대, 진해구 580대, 성산구 506대, 마산합포구 453대 순으로 확인
이는 업체 수와 실제 운송 공급력이 반드시 일치하지 않음을 보여줌


항만 인접지역인 마산합포구와 진해구에는 총 208개 업체와 1,033대의 차량이 분포한 반면, 의창구·성산구·마산회원구 등 내륙·산단 배후지역에는 358개 업체와 2,461대의 차량이 분포, 따라서 창원시의 화물차 운송 공급력은 항만 인접지역보다 내륙 및 산업 배후지역에 더 집중되어 있는 것으로 나타남, 이는 향후 항만과 산업단지, 운송사업자 간 공차 이동을 줄이기 위한 귀로 화물 매칭 모델 구축의 필요성을 뒷받침함